C2GUILauncher/src/Mods/ModManager.cs

```diff
@@ -0,0 +1,156 @@
+                try
+                {
+                    // Attempt to get master branch
+                    return await HttpHelpers.GetRawContentsAsync(repoRoot + "/main/mod.json");
+                }
+                // If it 404s
+                catch (HttpRequestException e) when (e.StatusCode == HttpStatusCode.NotFound)
+                {
+                    // Attempt to get main branch instead
+                    return await HttpHelpers.GetRawContentsAsync(repoRoot + "/master/mod.json");
+
+                }
+            }
+            catch (Exception e)
+            {
+                // TODO: More robust error handling. This should arccumulate the failures and report them to the UI somehow.
+                throw new Exception($"Failed to download {repoRoot} mod json.", e);
+            }
+        }
+
+        public IEnumerable<DownloadTask> DownloadModFiles(bool debug)
+        {
+            // Create plugins dir. This method does nothing if the directory already exists.
+            Directory.CreateDirectory(CoreMods.PluginPath);
+
+            // All Chiv2-Community dll releases have an optional _dbg suffix for debug builds.
+            var downloadFileSuffix = debug ? "_dbg.dll" : ".dll";
+
+            // These are the core mods necessary for asset loading, server hosting, server browser usage, and the injector itself.
+            // Please forgive the jank debug dll implementation. It'll be less jank after we aren't using hardcoded paths
+            var coreMods = new List<DownloadTarget>() {
+                new DownloadTarget(CoreMods.AssetLoaderPluginURL.Replace(".dll", downloadFileSuffix), CoreMods.AssetLoaderPluginPath),
+                new DownloadTarget(CoreMods.ServerPluginURL.Replace(".dll", downloadFileSuffix), CoreMods.ServerPluginPath),
+                new DownloadTarget(CoreMods.BrowserPluginURL.Replace(".dll", downloadFileSuffix), CoreMods.BrowserPluginPath)
+            };
+
+            return HttpHelpers.DownloadAllFiles(coreMods);
+        }
+    }
+
+    [JsonConverter(typeof(StringEnumConverter))]
+    enum ModType
+    {
+        [System.Runtime.Serialization.EnumMember(Value = "PLUGIN")]
+        Plugin,
+
+        [System.Runtime.Serialization.EnumMember(Value = "HOST_ONLY_PLUGIN")]
+        HostOnlyPlugin,
+
+        [System.Runtime.Serialization.EnumMember(Value = "CLIENT_ONLY_PLUGIN")]
+        ClientOnlyPlugin,
+
+        [System.Runtime.Serialization.EnumMember(Value = "ASSETS")]
+        Assets
+    }
+    record Mod(
+        [property: JsonProperty("name")] string Name,
+        [property: JsonProperty("description")] string Description,
+        [property: JsonProperty("image_url")] string ImageUrl,
+        [property: JsonProperty("download_url")] string DownloadUrl,
+        [property: JsonProperty("mod_type")] string ModType,
+        [property: JsonProperty("authors")] List<string> Authors,
+        [property: JsonProperty("versions")] List<string>? Versions
+    );
```

Comment:
This is where the model that loads the mod json is defined. The enum is unused right now, but will be used on mod_type in the json.